woocommerce/woocommerce-android
# Add product: Show celebratory view when the first product is created

Part of [#9080](https://github.com/woocommerce/woocommerce-android/issues/9080)


### Description


This PR adds a celebration dialog when merchant publishes a product for the first time. The dialog offers the option to share the product:

<img src="https://github.com/woocommerce/woocommerce-android/assets/266376/f3758a0e-901a-416d-8a53-f66cdc633680" width="42%" />

For non-public sites (for example a free trial WooExpress site), the "Share product" button above will be hidden since it doesn't make sense to share a URL to a non-public site.

### Testing instructions


1. Start with a new store that has no products before,
2. Create a new product and publish it,
3. Ensure the "First product created" dialog is shown.
4. Test the "Share product" button,
5. Test the "Dismiss" button.

**Tips:** The celebration dialog will always appear whenever the site previously had zero products. So, to trigger the dialog multiple times, you can also publish then delete, then publish another.

Also worth testing on a non-public site to ensure the "Share product" button is not shown in this situation.
